src/webui/api/torrentscontroller.cpp

```diff
@@ -1143,3 +1143,46 @@ void TorrentsController::tagsAction()
         result << tag;
     setResult(result);
 }
+
+void TorrentsController::renameFileAction()
+{
+    checkParams({"hash", "id", "name"});
+
+    const QString hash = params()["hash"];
+    BitTorrent::TorrentHandle *const torrent = BitTorrent::Session::instance()->findTorrent(hash);
+    if (!torrent)
+        throw APIError(APIErrorType::NotFound);
```

Comment:
You are still continuing this pernicious practice of throwing error in such cases?.. Have you forgotten all our discussions and the conclusions we have drawn? Or do you want to redo them all at once later, when we decide on the format of the response?

Reply:
I'd rather redo them all at once